website/pages/blog/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset.mdx

```diff
@@ -0,0 +1,436 @@
+OK, I am exaggerating here but the experience was very traumatising.
+
+Guidelines alone is only good if people follow it. Without explicit and automatic enforcement, things start to slip and bad practices spread sooner or later. 
+
+To enforce guidelines, we need tools. Luckily, these days we have an extensive range of tools for best practices a GraphQL server:
+
+*   [GraphQL Codegen](https://the-guild.dev/graphql/codegen) with [typescript](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript) and [typescript-resolvers](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-resolvers) plugins for type-safe GraphQL server development in TypeScript
+*   [GraphQL ESLint](https://github.com/B2o5T/graphql-eslint) for validation and conventions like naming and best practices.
+
+However, there are still areas to improve.
+
+For example, the guideline for one of the GraphQL servers I am working on is to use generated types from GraphQL Codegen. However, some people do not follow the guidelines, mostly because they are not aware of it. Luckily, other team members or the Maintainers catch these issues at PR review time. But it could have saved everyone time and effort if we had tools that could enforce this guideline.
+
+### 2. How to Minimise Noise to the Maintainers
+
+Maintainers usually need to manage these aspects of a GraphQL server:
+
+*   Core server logic: resolver map, schemas, CI/CD, etc. 
+*   Config files: `.graphqlrc`, `codegen.yml`, etc.
+
+Changing anything in the Maintainers' domain should notify them. Unfortunately, this happens regularly in normal workflows. Maintainers also have their team's work and general package and security updates to worry about. So, being notified on unrelated PRs quickly lead to burnout.
+
+For example, if a new resolver is added, they must be manually added to the resolver map. So, Maintainers are notified on every new resolver. It may look like this if you are using [GraphQL Yoga](https://the-guild.dev/graphql/yoga-server):
+
+```ts
+// server.ts
+import { createServer } from 'node:http'
+import { createSchema, createYoga } from 'graphql-yoga'
+import { Auth } from "./schema/auth/resolvers";
+import { book, Book } from "./schema/auth/resolvers";
+import { user, User } from "./schema/user/resolvers";
+
+const schema = createSchema({
+  typeDefs: `...`,
+  resolvers: { // This is the resolver map
+    Query: {
+      user,
+      book,
+    },
+    Auth,
+    Book
+    User,
+  }
+});
+
+const yoga = createYoga({ schema });
+const server = createServer(yoga)
+server.listen(4000, () => {
+  console.info('Server is running on http://localhost:4000/graphql')
+})
+```
+
+There is a way to mitigate this issue: each schema module exports an object of resolvers and they are merged with [mergeResolvers from @graphql-tools/merge](https://the-guild.dev/graphql/tools/docs/schema-merging#merging-resolvers). This means the Maintainers are notified on every new module instead of every resolver:
```

Comment:
**Thought:** Are we ok with calling out the caveats of `mergeResolvers` from `@graphql-tools/merge` here?

It's a great tool as I use it at work too but I can also see the caveat in the `graphql-tools` doc so maybe it's fine here? I have added a gif, hopefully it softens it and make it a bit less serious :)